frontend/src/containers/Addgroup.js

```diff
@@ -0,0 +1,392 @@
+                          maxLength={50}
+                          placeholder="Charlie Chaplin"/>
+                      </div>
+                      <div>
+                        <div className="Addgroup-label">Email</div>
+                         <Input
+                          name={'email'}
+                          value={users4.email}
+                          onChange={(value) => appendGroupForm({users4: {name: users4.name, email: value, twitterHandle: users4.twitterHandle}})}
+                          maxLength={50}
+                          placeholder="charlie@chaps.com"/>
+                      </div>
+                      <div>
+                        <div className="Addgroup-label">Twitterhandle </div>
+                         <Input
+                          name={'twitterHandle'}
+                          value={users4.twitterHandle}
+                          onChange={(value) => appendGroupForm({users4: {name: users4.name, email: users4.email, twitterHandle: value}})}
+                          maxLength={50}
+                          placeholder="@charliechaps"/>
+                      </div>
+                    </div>
+                    <div className="flex flex-row">
+                      <div>
+                        <div className="Addgroup-label">Name</div>
+                        <Input
+                          name={'name'}
+                          value={users5.name}
+                          onChange={(value) => appendGroupForm({users5: {name: value, email: users5.email, twitterHandle: users5.twitterHandle}})}
+                          maxLength={50}
+                          placeholder="Charlie Chaplin"/>
+                      </div>
+                      <div>
+                        <div className="Addgroup-label">Email</div>
+                         <Input
+                          name={'email'}
+                          value={users5.email}
+                          onChange={(value) => appendGroupForm({users5: {name: users5.name, email: value, twitterHandle: users5.twitterHandle}})}
+                          maxLength={50}
+                          placeholder="charlie@chaps.com"/>
+                      </div>
+                      <div>
+                        <div className="Addgroup-label">Twitterhandle </div>
+                         <Input
+                          name={'twitterHandle'}
+                          value={users5.twitterHandle}
+                          onChange={(value) => appendGroupForm({users5: {name: users5.name, email: users5.email, twitterHandle: value}})}
+                          maxLength={50}
+                          placeholder="@charliechaps"/>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </div>
+              <div className={`Addgroup-Button`} onClick={this.create.bind(this)}>create!</div>
```

Comment:
`.bind(this)` In-lined, inside the `render` function is an anti-pattern, because a **new** function is created on every render.

In the contructor you can bind the method for use:

`this.createRef = this.create.bind(this);`

so now instead you can do:

`... onClick={ this.createRef }`





It is explained here:
https://medium.com/@esamatti/react-js-pure-render-performance-anti-pattern-fb88c101332f#.nsv25mbyg

**tl;dr**
> The anti-pattern is creating new arrays, objects, functions or any other new identities during render or in Redux connect(mapState).